src/routes/postSkipSegments.js

```diff
@@ -97,48 +97,26 @@ async function autoModerateSubmission(submission, callback) {
                 if (duration == 0) {
                     // Allow submission if the duration is 0 (bug in youtube api)
                     return false;
-                }
-
-                for (const segment of segments) {
-                    if ((segment.segment[1] - segment.segment[0]) > (duration / 100) * 80) {
-                        // Reject submission if over 80% of the video
-                        return "One of your submitted segments is over 80% of the video.";
+                } else if ((submission.endTime - submission.startTime) > (duration/100)*80) {
+                    // Reject submission if over 80% of the video
+                    return "Sponsor segment is over 80% of the video.";
+                } else {
+                    // Check NeuralBlock
+                    let neuralBlockURL = config.neuralBlockURL;
+                    if (!neuralBlockURL) return false;
+
+                    let response = await fetch(neuralBlockURL + "/api/checkSponsorSegments?vid=" + submission.videoID +
+                                              "&segments=" + submission.startTime + "," + submission.endTime);
+                    if (!response.ok) return false;
+
+                    let nbPredictions = await response.json();
+                    if (nbPredictions.probabilities[0] >= 0.70){
```

Comment:
The main check now occurs here and is set at 70%. Some testing I think will be needed to determine the best threshold. Also we still need to forward the rejection to discord for manual review

Reply:
We can log the percentage in the discord message as well